Some compilation errors after the last commit b70b0f1a0b76a3fba6fb62f7eadce327be8e9203

I don't know if it is ogre or my project but in my personal project here I got this compilation error,
compiling with clang 22.1.6, ogre3d compiled with this same version.
```
In file included from /usr/local/include/OGRE/Bullet/OgreBullet.h:12:
In file included from /usr/local/include/OGRE/Ogre.h:31:
In file included from /usr/local/include/OGRE/OgrePrerequisites.h:327:
In file included from /usr/local/include/OGRE/OgreStdHeaders.h:13:
In file included from /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/vector:68:
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_vector.h:1109:44: error:
      arithmetic on a pointer to an incomplete type 'Ogre::MeshLodUsage'
 1109 |         ptrdiff_t __dif = this->_M_impl._M_finish - this->_M_imp...
      |                           ~~~~~~~~~~~~~~~~~~~~~~~ ^
/usr/local/include/OGRE/OgreMesh.h:495:71: note: in instantiation of member
      function 'std::vector<Ogre::MeshLodUsage>::size' requested here
  495 |   ...getNumLodLevels(void) const { return mMeshLodUsageList.size(); }
      |                                                             ^
/usr/local/include/OGRE/OgreMesh.h:54:12: note: forward declaration of
      'Ogre::MeshLodUsage'
   54 |     struct MeshLodUsage;
      |            ^
1 error generated.
make[2]: *** [CMakeFiles/dda.dir/build.make:79: CMakeFiles/dda.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/dda.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

--- a/OgreMain/include/OgreMesh.h
+++ b/OgreMain/include/OgreMesh.h
@@ -30,6 +30,8 @@ THE SOFTWARE.
 
 #include "OgrePrerequisites.h"
 
+#include <vector>
+
 #include "OgreResource.h"
 #include "OgreAxisAlignedBox.h"
 #include "OgreVertexBoneAssignment.h"

--- a/PlugIns/DotScene/include/OgreDotSceneLoader.h
+++ b/PlugIns/DotScene/include/OgreDotSceneLoader.h
@@ -87,8 +87,8 @@ class _OgreDotScenePluginExport DotScenePlugin : public Plugin
 
     void install() override;
     void initialise() override {}
-    void shutdown() override;
-    void uninstall() override {}
+    void uninstall() override;
+    void shutdown() override {}
 private:
     Codec* mCodec;
 };

--- a/PlugIns/DotScene/src/DotSceneLoader.cpp
+++ b/PlugIns/DotScene/src/DotSceneLoader.cpp
@@ -1143,7 +1143,7 @@ void DotScenePlugin::install()
     Codec::registerCodec(mCodec);
 }
 
-void DotScenePlugin::shutdown() {
+void DotScenePlugin::uninstall() {
     Codec::unregisterCodec(mCodec);
     delete mCodec;
 }

--- a/PlugIns/STBICodec/src/OgreSTBICodec.cpp
+++ b/PlugIns/STBICodec/src/OgreSTBICodec.cpp
@@ -178,9 +178,9 @@ namespace Ogre {
 
         if (!pixelData)
         {
-            OGRE_EXCEPT(Exception::ERR_INTERNAL_ERROR, 
-                "Error decoding image: " + String(stbi_failure_reason()),
-                "STBIImageCodec::decode");
+            const char* reason = stbi_failure_reason();
+            OGRE_EXCEPT(Exception::ERR_INTERNAL_ERROR,
+                        "Error decoding image: " + String(reason ? reason : "no reason"));
         }
 
         if (width <= 0 || height <= 0)

